src/modules/auth/jwt.strategy.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock } from '@golevelup/ts-jest';
+import { UserUseCases } from '../user/user.usecase';
+import { JwtStrategy } from './jwt.strategy';
+import { newUser } from '../../../test/fixtures';
+import { ConfigService } from '@nestjs/config';
+import { UnauthorizedException } from '@nestjs/common';
+import { getTokenDefaultIat } from '../../lib/jwt';
+
+describe('Jwt strategy', () => {
+  let userUseCases: UserUseCases;
+  let strategy: JwtStrategy;
+
+  beforeEach(async () => {
+    const moduleRef = await createTestingModule();
+    userUseCases = moduleRef.get<UserUseCases>(UserUseCases);
+    strategy = moduleRef.get<JwtStrategy>(JwtStrategy);
+  });
+
+  it('When token is old, then fail', async () => {
+    await expect(strategy.validate({ email: 'test@test.com' })).rejects.toThrow(
+      new UnauthorizedException('Old token version detected'),
+    );
+  });
+
+  it('When user does not exist, then fail', async () => {
+    jest.spyOn(userUseCases, 'getUser').mockResolvedValue(null);
+
+    await expect(
+      strategy.validate({ payload: { uuid: 'anyUuid' } }),
+    ).rejects.toThrow(UnauthorizedException);
+  });
+
+  it('When token iat is older than lastPasswordChangedAt , then fail', async () => {
+    const user = newUser();
+    const greaterDate = new Date();
+    greaterDate.setMinutes(greaterDate.getMinutes() + 1);
+    user.lastPasswordChangedAt = greaterDate;
+    const tokenIat = getTokenDefaultIat();
+
+    jest.spyOn(userUseCases, 'getUser').mockResolvedValue(user);
+
+    await expect(
+      strategy.validate({ payload: { uuid: 'anyUuid' }, iat: tokenIat }),
+    ).rejects.toThrow(UnauthorizedException);
+  });
+
+  it('When user has lastPasswordChangedAt older than token iat, then return user', async () => {
+    const tokenIat = getTokenDefaultIat();
+    const user = newUser();
+    const olderDate = new Date();
```

Comment:
Same here. Switching the orders of both things could make the tests pass. The control of the things that determine if the tests passes or not should be clear here. Ideally, we should not have to check the internal code to be sure that the tests works as expected